### PR TITLE
remove py-solc-x

### PR DESCRIPTION
**Pull request by BitBaseBit**

### CODE REVIEW

File: warp/cairo-src/evm/calls.cairo [hunk 1]
```diff
@@ -56,10 +56,10 @@ func calldata_load{range_check_ptr, exec_env : ExecutionEnvironment*}(offset) ->
     return (value=value)
 end
 
-func returndata_copy{range_check_ptr, exec_env : ExecutionEnvironment, memory_dict : DictAccess*}(
+func returndata_copy{range_check_ptr, exec_env : ExecutionEnvironment*, memory_dict : DictAccess*}(
         memory_pos : Uint256, returndata_pos : Uint256, length : Uint256):
     array_copy_to_memory(
-        exec_env.returndata_len, exec_env.returndata, returndata_pos, memory_pos, length)
+        exec_env.returndata_len, exec_env.returndata, returndata_pos.low, memory_pos.low, length.low)
```
Comment (murcake) [inline warp/cairo-src/evm/calls.cairo:62]: Let's do it in a separate commit and add a test to ensure that desync doesn't happen again

File: warp/yul/main.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def transpile_from_yul(
     yul_ast = SwitchToIfVisitor().map(yul_ast)
     yul_ast = ExpressionSplitter(name_gen).map(yul_ast)
     yul_ast = RevertNormalizer().map(yul_ast)
-    yul_ast = ScopeFlattener(name_gen).map(yul_ast)
+    # yul_ast = ScopeFlattener(name_gen).map(yul_ast)
```
Comment (murcake) [inline warp/yul/main.py:89]: Let's also remove ScopeFlattener in another commit

File: requirements.txt [hunk 3]
```diff
@@ -6,7 +6,6 @@ sympy
 pytest==6.2.4
 pytest-parallel
```
Comment (murcake) [inline requirements.txt:7]: Can't comment on binary files, so commenting here =(. Why is the linux/kudu binary moved from `./warp/bin` to `./bin`? I think setup.py expects it to be in the former directory?

**Review by murcake — APPROVED** [review]
It's nice to get rid of another dependency

**Comment by murcake** [comment]
I see that some tests fail, because we attempt to take the low part of a Uint256 constructor. #122 should help with that

**Comment by BitBaseBit** [comment]
> I see that some tests fail, because we attempt to take the low part of a Uint256 constructor. #122 should help with that

This is odd because this test passes locally.

**Comment by murcake** [comment]
> > I see that some tests fail, because we attempt to take the low part of a Uint256 constructor. #122 should help with that
> 
> This is odd because this test passes locally.

Do you run make test or make test_yul? The latter doesn't run every test

**Comment by BitBaseBit** [comment]
> > > I see that some tests fail, because we attempt to take the low part of a Uint256 constructor. #122 should help with that
> > 
> > 
> > This is odd because this test passes locally.
> 
> Do you run make test or make test_yul? The latter doesn't run every test

Ah okay , yeah the issue is there when I run test_yul

**Comment by BitBaseBit** [comment]
> > > I see that some tests fail, because we attempt to take the low part of a Uint256 constructor. #122 should help with that
> > 
> > 
> > This is odd because this test passes locally.
> 
> Do you run make test or make test_yul? The latter doesn't run every test

Go ahead & merge #122

**Comment by murcake** [comment]
(also, it seems that this version of kudu is expression-splitting and the pr for it is #119 AFAIU)
